src/Avalonia.Controls/SplitButton/SplitButton.cs

```diff
@@ -0,0 +1,709 @@
+            set => SetValue(CommandParameterProperty, value);
+        }
+
+        /// <summary>
+        /// Gets or sets the <see cref="FlyoutBase"/> that is shown when the secondary part is pressed.
+        /// </summary>
+        public FlyoutBase Flyout
+        {
+            get => GetValue(FlyoutProperty);
+            set => SetValue(FlyoutProperty, value);
+        }
+
+        /// <summary>
+        /// Gets a value indicating whether the button is currently checked.
+        /// </summary>
+        /// <remarks>
+        /// This property exists only for the derived <see cref="ToggleSplitButton"/> and is
+        /// unused (set to false) within <see cref="SplitButton"/>. Doing this allows the
+        /// two controls to share a default style.
+        /// </remarks>
+        internal virtual bool InternalIsChecked => false;
+
+        /// <inheritdoc/>
+        protected override bool IsEnabledCore => base.IsEnabledCore && _commandCanExecute;
+
+        ////////////////////////////////////////////////////////////////////////
+        // 
+        // Methods
+        //
+        ////////////////////////////////////////////////////////////////////////
+
+        /// <inheritdoc/>
+        public void CanExecuteChanged(object sender, EventArgs e)
+        {
+            var canExecute = Command == null || Command.CanExecute(CommandParameter);
+
+            if (canExecute != _commandCanExecute)
+            {
+                _commandCanExecute = canExecute;
+                UpdateIsEffectivelyEnabled();
+            }
+        }
+
+        /// <summary>
+        /// Updates the visual state of the control by applying latest PseudoClasses.
+        /// </summary>
+        protected void UpdatePseudoClasses()
+        {
+            string pcNormal   = ":normal"; // Not supported in XAML style
+            string pcDisabled = ":disabled";
+
+            string pcSecondaryButtonRight = ":secondaryButtonRight";
+            string pcSecondaryButtonSpan  = ":secondaryButtonSpan";
+
+            string pcCheckedFlyoutOpen = ":checkedFlyoutOpen";
+            string pcFlyoutOpen        = ":flyoutOpen";
+
+            string pcCheckedTouchPressed         = ":checkedTouchPressed";
+            string pcChecked                     = ":checked";
+            string pcCheckedPrimaryPressed       = ":checkedPrimaryPressed";
+            string pcCheckedPrimaryPointerOver   = ":checkedPrimaryPointerOver";
+            string pcCheckedSecondaryPressed     = ":checkedSecondaryPressed";
+            string pcCheckedSecondaryPointerOver = ":checkedSecondaryPointerOver";
+
+            string pcTouchPressed         = ":touchPressed";
+            string pcPrimaryPressed       = ":primaryPressed";
+            string pcPrimaryPointerOver   = ":primaryPointerOver";
+            string pcSecondaryPressed     = ":secondaryPressed";
+            string pcSecondaryPointerOver = ":secondaryPointerOver";
```

Comment:
It might be better to 'const' these locally or move them out of the method and make them readonly/const so they're not created every time pseudoclasses need to be updated. Also probably don't need the `:normal` class, that can just be the default style itself

Reply:
> It might be better to 'const' these locally or move them out of the method and make them readonly/const 

Good point, will definitely move these out.

> `:normal` is still used in code for readability and to match WinUI. However, in the local function that actually sets the pseudoclasses it won't match with anything and therefore won't matter (all classes are cleared in that case). I could probably delete this since I'm diverging from WinUI already.

Reply:
Should be resolved:
 1. I made all these class constants now. This also allows me to use them in the PseudoClasses class attribute which removes the last magic strings in code-behind. So we have a 2-for-1 here :)
 2. I removed the ':normal' PseudoClass and just leave the default empty string which has the same functionality (will clear all other exclusive PseudoClasses)